### ISSUE TITLE
`mdbook serve` causes rebuild loop

### ISSUE THREAD

**Issue by htngr** [danieleades/mdbook-d2#16]
If I run `mdbook serve` with mdbook-d2 as a preprocessor and include a d2 diagram in my book, mdbook will constantly rebuild because d2 outputs diagrams in `src/d2`:
```
2023-02-08 10:58:04 [INFO] (mdbook::cmd::serve): Files changed: ["book/src/file.md"]
2023-02-08 10:58:04 [INFO] (mdbook::cmd::serve): Building book...
2023-02-08 10:58:04 [INFO] (mdbook::book): Book building has started
2023-02-08 10:58:04 [INFO] (mdbook::book): Running the html backend
2023-02-08 10:58:05 [INFO] (mdbook::cmd::serve): Files changed: ["book/src/d2/2.1.svg"]
2023-02-08 10:58:05 [INFO] (mdbook::cmd::serve): Building book...
2023-02-08 10:58:05 [INFO] (mdbook::book): Book building has started
2023-02-08 10:58:06 [INFO] (mdbook::book): Running the html backend
2023-02-08 10:58:07 [INFO] (mdbook::cmd::serve): Files changed: ["book/src/d2/2.1.svg"]
2023-02-08 10:58:07 [INFO] (mdbook::cmd::serve): Building book...
2023-02-08 10:58:07 [INFO] (mdbook::book): Book building has started
2023-02-08 10:58:07 [INFO] (mdbook::book): Running the html backend
2023-02-08 10:58:08 [INFO] (mdbook::cmd::serve): Files changed: ["book/src/d2/2.1.svg"]
2023-02-08 10:58:08 [INFO] (mdbook::cmd::serve): Building book...
2023-02-08 10:58:08 [INFO] (mdbook::book): Book building has started
2023-02-08 10:58:08 [INFO] (mdbook::book): Running the html backend
...
```

Adding `src/d2` to `.gitignore` doesn't work, because `mdbook serve` somehow ignores this. When changing the d2 output dir to `../d2` mdbook cant find the generated diagrams.

Maybe this can be solved by #8.

**Comment by danieleades** [comment]
this is caused by an upstream bug here - https://github.com/rust-lang/mdBook/issues/1186

#8 would be a workaround, rather than a solution, as PDF output would no longer be supported. Still, worth adding support for inline SVGs.
doesn't look like #10 is being worked on anymore, so I'll take a look at picking that up myself

**Comment by danieleades** [comment]
I'm going to close this, since it's the result of an upstream bug, and a workaround was implemented in #17

**Comment by htngr** [comment]
Thanks a lot :muscle:
